tensorflow/python/ops/nn_grad.py

```diff
@@ -203,6 +204,27 @@ def _BiasAddGrad(op, received_grad):
   return (received_grad, gen_nn_ops.bias_add_grad(out_backprop=received_grad,
                                                   data_format=data_format))
 
+@ops.RegisterGradient("BiasAddGrad")
+def _BiasAddGradGrad(op, received_grad):
+  """Gradient for the BiasAddGrad op.
+
+  Args:
+    op: BiasAddGrad op for which we are calculating gradients.
+    received_grad: The gradients passed to the BiasAddGrad op.
+    
+  Returns:
+    A single gradient Tensor for the input to BiasAddGrad (which
+    is the gradient of the bias term in BiasAdd)
+  """
+  
+  try:
+    data_format = op.get_attr("data_format")
+  except ValueError:
+    data_format = None
+    
+  zeros = gen_array_ops._zeros_like(op.inputs[0])
+  return gen_nn_ops._bias_add(zeros, received_grad, data_format=data_format)
```

Comment:
In cases where the shapes are known, I think a broadcasting op such as tf.tile will be better. Since it avoids allocating the zero input array and add the unnecessary zeros. However, tf.tile in its current form may be slower. So a bit benchmarking is needed for that path. I'll leave it up to you whether you want to try that. 

Also we need to add a test using compute_gradient_errort to make sure this passes the gradient checker.